service/src/main/java/com/google/auto/service/processor/AutoServiceProcessor.java

```diff
@@ -243,23 +251,16 @@ private String getBinaryNameImpl(TypeElement element, String className) {
     return getBinaryNameImpl(typeElement, typeElement.getSimpleName() + "$" + className);
   }
 
-  private DeclaredType getProviderInterface(AnnotationMirror providerAnnotation) {
-
-    // The very simplest of way of doing this, is also unfortunately unworkable.
-    // We'd like to do:
-    //    ServiceProvider provider = e.getAnnotation(ServiceProvider.class);
-    //    Class<?> providerInterface = provider.value();
-    //
-    // but unfortunately we can't load the arbitrary class at annotation
-    // processing time. So, instead, we have to use the mirror to get at the
-    // value (much more painful).
-
-    Map<? extends ExecutableElement, ? extends AnnotationValue> valueIndex =
-        providerAnnotation.getElementValues();
-    log("annotation values: " + valueIndex);
-
-    AnnotationValue value = valueIndex.values().iterator().next();
-    return (DeclaredType) value.getValue();
+  /**
+   * Returns the contents of a {@code Class[]}-typed "value" field in a given {@code annotationMirror}.
+   */
+  private ImmutableSet<DeclaredType> getValueFieldOfClasses(AnnotationMirror annotationMirror) {
+    //noinspection unchecked
```

Comment:
nit: remove

Reply:
Ah, I didn't realize what you were doing. I'll just change it to a regular suppress warnings

Reply:
:+1: